### PR TITLE
*:rechange to retry when tikv disk full

### PR DESCRIPTION
**Pull request by tier-cap**
Signed-off-by: tier-cap <zhengxiaojin@pingcap.com>
### What problem does this PR solve?
1. When TiKV is disk full, the original design is just return the disk full error to user without any retry.
2. Currently, need change to retry before return to user.

### What is changed and how it works?
1. Use backoff to retry it when TiKV is disk full.

### CODE REVIEW

File: internal/unionstore/memdb.go [hunk 1]
```diff
@@ -166,10 +165,10 @@ func (db *MemDB) Reset() {
...
 }
 
-// SetDiskFullOpt is used by TiDB test case.
-func (db *MemDB) SetDiskFullOpt(level kvrpcpb.DiskFullOpt) {
-	// Nothing to do.
-}
+// // SetDiskFullOpt is used by TiDB test case.
+// func (db *MemDB) SetDiskFullOpt(level kvrpcpb.DiskFullOpt) {
+// 	// Nothing to do.
+// }
```
Comment (sticnarf) [inline internal/unionstore/memdb.go:171]: Please remove the code.

Comment (tier-cap) [reply internal/unionstore/memdb.go:171]: Right, tidb has already removed this func.
Done!

File: internal/retry/config.go [hunk 2]
```diff
@@ -99,6 +99,10 @@ func (c *Config) String() string {
 	return c.name
 }
 
+func (c *Config) SetErrors(err error) {
```
Comment (disksing) [inline internal/retry/config.go:102]: please add comment. the rest LGTM

Comment (tier-cap) [reply internal/retry/config.go:103]: OK, Done

**Review by disksing — APPROVED** [review]

**Pull request merged**